### DISCUSSION TITLE
Release of ezdxf v1.0.0

### DISCUSSION THREAD

**Discussion by mozman** [mozman/ezdxf#792] [Announcements]
Release of ezdxf v1.0.0
=====

The project started in 2010 as the next generation of the `dxfwrite` package that added loading and editing features and implemented support for DXF versions newer than R12, and has now finally reached version 1.0 status.

A big thanks to @mbway for contributing the `drawing` add-on which was the reason for adding a general transform interface and the path tools. These additions have greatly improved `ezdxf` as many tools are based on these basic features: the `explode`, `disassemble`, `bbox` and `zoom` modules, the `text2path` and `geo` add-ons, and many internal features. And the `drawing` add-on itself seems to be one of the most used features of the entire package - judging by the number of questions and bug reports associated with it 😉. 

And thanks to all people who reported bugs, especially @chibai!

Version 1.0.0 is basically version 0.18.1 with the deprecated functions and methods removed and a major revision of the documentation and type annotations. For more information about new features and changes go to the [release page](https://ezdxf.mozman.at/release-v1-0.html).

Upcoming
-----

There will be an extended period of only bug fixes without adding any major new features that will be released as a 1.0.x series to take a break from the project and these versions will not change any APIs, dependencies or minimum requirements. For version 1.1, the minimum Python version will be raised to 3.8 or possibly 3.9 since even the latest LTS version of Ubuntu ships with Python 3.9.
Or I sync the minimal Python version and provided binary wheels on PyPI with *matplotlib*, which is still releasing binary wheels for Python 3.8 at the end of 2022.

As for the new features, there are already many ideas in the *TODO.md* file, but I haven't decided yet which ones will be added and when, as this is still a hobby project, there is no set schedule for future releases: It's done when it's done.

It's also time to expand the DXF-Internals documentation to bring the knowledge baked into code into a more readable form.

Please post any comments, ideas or suggestions here in the discussion forum.

See you sometime, take care, stay healthy!

**Comment by chibai** [comment]
Wa~~It's my pleasure!!
